packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -1,53 +1,161 @@
+import {
+  changePassword,
+  runFieldValidators,
+  translate,
+  FieldValidator,
+  getDefaultPasswordValidators,
+  getConfirmPasswordValidator,
+  FormEventType,
+} from '@aws-amplify/ui';
 
 import { useAuth } from '../../../internal';
 import { View, Flex } from '../../../primitives';
+import { FormValues, BlurredFields, ValidationError } from '../types';
 import {
+  DefaultConfirmPassword,
   DefaultCurrentPassword,
   DefaultError,
   DefaultNewPassword,
   DefaultSubmitButton,
 } from './defaultComponents';
 import { ChangePasswordProps } from './types';
-import { FormValues } from '../types';
 
 const logger = new Logger('ChangePassword');
 
+const getIsDisabled = (
+  formValues: FormValues,
+  validationError: ValidationError
+): boolean => {
+  const { currentPassword, newPassword, confirmPassword } = formValues;
+
+  if (!currentPassword || !newPassword || !confirmPassword) {
+    // if passwords aren't entered yet, disable submit
+    return true;
+  } else if (
+    // if there are some password validation error, disable submit
+    validationError.newPassword?.length > 0 ||
+    validationError.confirmPassword?.length > 0
+  ) {
+    return true;
+  }
+
+  return false;
+};
+
 function ChangePassword({
   onSuccess,
   onError,
+  validators,
 }: ChangePasswordProps): JSX.Element | null {
   const [errorMessage, setErrorMessage] = React.useState<string>(null);
   const [formValues, setFormValues] = React.useState<FormValues>({});
-
+  const [blurredFields, setBlurredFields] = React.useState<BlurredFields>({});
+  const [validationError, setValidationError] = React.useState<ValidationError>(
+    {}
+  );
   const { user, isLoading } = useAuth();
 
-  /** Return null if Auth.getCurrentAuthenticatedUser is still in progress  */
-  if (isLoading) {
-    return null;
-  }
+  const isDisabled = getIsDisabled(formValues, validationError);
 
-  /** Return null if user isn't authenticated in the first place */
-  if (!user) {
-    logger.warn('<ChangePassword /> requires user to be authenticated.');
-    return null;
-  }
+  const passwordValidators: FieldValidator[] = React.useMemo(() => {
+    return validators ?? getDefaultPasswordValidators();
+  }, [validators]);
+
+  const validateNewPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: FormEventType
+    ): string[] => {
+      const { newPassword } = formValues;
+
+      return runFieldValidators(
+        newPassword,
+        passwordValidators,
+        eventType,
+        blurredFields.newPassword
+      );
+    },
+    [passwordValidators]
+  );
+
+  const validateConfirmPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: FormEventType
+    ): string[] => {
+      const { newPassword, confirmPassword } = formValues;
+
+      const confirmPasswordValidator = getConfirmPasswordValidator(newPassword);
```

Comment:
`confirmPasswordValidator` is getting generated dynamically, since it depends on `confirmPassword`.